src/Marten/Events/EventStore.cs

```diff
@@ -245,8 +247,9 @@ public async Task AppendExclusive(string streamKey, CancellationToken token, par
             _store.Events.EnsureAsStringStorage(_session);
             await _session.Database.EnsureStorageExistsAsync(typeof(IEvent), token).ConfigureAwait(false);
 
-            var cmd = new NpgsqlCommand($"select version from {_store.Events.DatabaseSchemaName}.mt_streams where id = :id for update")
-                .With("id", streamKey);
+            var cmd = new NpgsqlCommand($"select version from {_store.Events.DatabaseSchemaName}.mt_streams where id = :id and tenant_id = :tenant_id for update")
```

Comment:
This is a good catch, but we'll have to do some conditional logic here and only use the tenant_id version of the command when the event store is configured to be multi-tenanted. The tenant_id column only exists in that case.

Reply:
It is, got it, and getting it applied with just a little structural improvement to the bigger EventSTore class that's gotten too big. Thank you so much for doing this, I pulled it down before I noticed you'd made the fix. This will be in 5.1.0 sometime in the next week or so.